src/test/java/julia/git/CommitTest.java

```diff
@@ -45,4 +45,50 @@ public void possibleToGetHashOfCommit() {
         Commit commit = new Commit(directory.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
         assertEquals("932af51418307f9edf6387a183cebc8735c34c3e", commit.getHash());
     }
+
+    @Test
+    public void hashOfCommitIsChangedWhenParentIsChanged() {
```

Comment:
Need to create a new commit based on the old commit - that's the only way to get the "same" commit with a different parent.